### PR TITLE
Add --file flag to 'minikube logs' to automatically put logs into a file.

### LINKED ISSUE #11231: Add --file flag to minikube logs.
We should have a --file flag in `minikube logs` so that logs output is sent directly to a file for easy transfer.

Before:
```
$ minikube logs
<tons of output>
```
After:
```
$ minikube logs --file out
```
File 'out' written full of logs.

### PR DESCRIPTION
**Pull request by andriyDev**
fixes #11231.

Before:
```
$ minikube logs

==> Audit <==
|---------------|--------------------------------------------|----------|-------------|---------|-------------------------------|-------------------------------|
|    Command    |                    Args                    | Profile  |    User     | Version |          Start Time           |           End Time            |
|---------------|--------------------------------------------|----------|-------------|---------|-------------------------------|-------------------------------|
| generate-docs | --path ./site/content/en/docs/commands/    | minikube | andriydzikh | v1.19.0 | Thu, 29 Apr 2021 13:58:18 PDT | Thu, 29 Apr 2021 13:58:18 PDT |
|               | --test-path                                |          |             |         |                               |                               |
|               | ./site/content/en/docs/contrib/tests.en.md |          |             |         |                               |                               |
| start         |                                            | minikube | andriydzikh | v1.19.0 | Thu, 29 Apr 2021 14:06:56 PDT | Thu, 29 Apr 2021 14:08:05 PDT |
... Continued
```
After:
```
andriydzikh@andriydzikh-glaptop:~/minikube$ minikube logs --file f
```
After file:
```
* 
* ==> Audit <==
* |---------------|--------------------------------------------|----------|-------------|---------|-------------------------------|-------------------------------|
|    Command    |                    Args                    | Profile  |    User     | Version |          Start Time           |           End Time            |
|---------------|--------------------------------------------|----------|-------------|---------|-------------------------------|-------------------------------|
| generate-docs | --path ./site/content/en/docs/commands/    | minikube | andriydzikh | v1.19.0 | Thu, 29 Apr 2021 13:58:18 PDT | Thu, 29 Apr 2021 13:58:18 PDT |
|               | --test-path                                |          |             |         |                               |                               |
|               | ./site/content/en/docs/contrib/tests.en.md |          |             |         |                               |                               |
| start         |                                            | minikube | andriydzikh | v1.19.0 | Thu, 29 Apr 2021 14:06:56 PDT | Thu, 29 Apr 2021 14:08:05 PDT |
... Continued
```

### CODE REVIEW

File: pkg/minikube/logs/logs.go [hunk 1]
```diff
@@ -93,16 +94,16 @@ type logRunner interface {
 const lookBackwardsCount = 400
 
 // Follow follows logs from multiple files in tail(1) format
-func Follow(r cruntime.Manager, bs bootstrapper.Bootstrapper, cfg config.ClusterConfig, cr logRunner) error {
+func Follow(r cruntime.Manager, bs bootstrapper.Bootstrapper, cfg config.ClusterConfig, cr logRunner, logOutput io.Writer) error {
```
Comment (medyagh) [inline pkg/minikube/logs/logs.go:97]: is this PR also fixing the Follow ?  minikube logs -f has not been working

Comment (andriyDev) [reply pkg/minikube/logs/logs.go:97]: I'm not sure how minikube logs -f is broken at the moment? As far as I can tell it's working properly on master.

**Comment by spowelljr** [comment]
@andriyDev Run `make generate-docs` to update the `--file` flag in the docs.

**Comment by andriyDev** [comment]
@spowelljr Done!

**Comment by medyagh** [comment]
@andriyDev so -file will still output to the console ? I would rather the -file only write to a file and let user's console be clean

**Comment by andriyDev** [comment]
> @andriyDev so -file will still output to the console ? I would rather the -file only write to a file and let user's console be clean

@medyagh No, the console is clean when using --file. Errors however will still output to the console though (e.g. permission denied). This way users can see any issues resulting from the actual logging process and not have that be tied into the log file.

File: cmd/minikube/cmd/logs.go [hunk 2]
```diff
@@ -52,7 +54,18 @@ var logsCmd = &cobra.Command{
...
-		logs.OutputOffline(numberOfLines)
+		var logOutput *os.File = os.Stdout
+		var err error
+
+		if fileOutput != "" {
+			logOutput, err = os.Create(fileOutput)
+			if err != nil {
+				exit.Error(reason.Usage, "Failed to create file", err)
+			}
+			defer logOutput.Close()
```
Comment (medyagh) [inline cmd/minikube/cmd/logs.go:65]: defer before exit

Comment (andriyDev) [reply cmd/minikube/cmd/logs.go]: Done.

File: cmd/minikube/cmd/logs.go [hunk 3]
```diff
@@ -52,7 +55,23 @@ var logsCmd = &cobra.Command{
 	Short: "Returns logs to debug a local Kubernetes cluster",
 	Long:  `Gets the logs of the running instance, used for debugging minikube, not user code.`,
 	Run: func(cmd *cobra.Command, args []string) {
-		logs.OutputOffline(numberOfLines)
+		var logOutput *os.File = os.Stdout
+		var err error
+
+		if fileOutput != "" {
+			logOutput, err = os.Create(fileOutput)
```
Comment (sharifelgamal) [inline cmd/minikube/cmd/logs.go:62]: So are we expecting an absolute or relative path here? Enforcing one or the other will make things less confusing for the user.

Comment (andriyDev) [reply cmd/minikube/cmd/logs.go:62]: It can be either. Isn't it standard to allow both relative and absolute paths for file parameters?

Comment (medyagh) [reply cmd/minikube/cmd/logs.go:62]: we should allow user both, I image I would wanna do
```
minikube logs -f ~/Desktop/m.log
```
or

```
minikube logs -f /tmp/m.log
```

**Comment by andriyDev** [comment]
Rebased onto master.

**Comment by andriyDev** [comment]
Rebased onto master #2.

**Comment by andriyDev** [comment]
Rebased onto master x3.

**Review by medyagh — APPROVED** [review]

**Comment by medyagh** [comment]
Looks good to me , Thank you for this PR and improving minikube,

**Pull request merged**